Release/v5.5.0

ErikSin opened the pull request.

ErikSin committed:
feat: Adds map setting screens (#946)

* feat: Added map settings

* chore: Translations

* feat: Created UX Pill Component

* Update en.json

* chore: Added Offline Area list

* chore: translations

* chore:completed BG Maps Screens

* chore: translations

* chore:handle import press

* chore:update Spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* Chore: Casing

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: PR fixes

* chore: translations

* chore: fixed flicker

* chore: modal components inline

* chore: updated Translations

* chore: hidden under feature flag

* chore: cleaned up feature flag

* chore: updated nav state feature flags

* feat: added experiments hook

* chore: feature flag clean up

* chore: clean up typing

* chore: clean up appContainer

* chore: clean up

* chore: cleaned up linting

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

achou11 committed:
Merge branch 'deploy' into develop

ErikSin committed:
Map selector (#966)

* feat: experimental: Added Map Settings

* chore: Translations

* feat: Created UX Pill Component

* Update en.json

* chore: Added Offline Area list

* chore: translations

* chore:completed BG Maps Screens

* chore: translations

* chore:handle import press

* chore:update Spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* Chore: Casing

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: PR fixes

* chore: translations

* chore: fixed flicker

* chore: modal components inline

* chore: updated Translations

* chore: hidden under feature flag

* chore: cleaned up feature flag

* chore: updated nav state feature flags

* feat: added experiments hook

* chore: feature flag clean up

* chore: clean up typing

* chore: clean up appContainer

* chore: clean up

* chore: cleaned up linting

* Added Background Map Selector to Map View

* chore: translations

* chore:Pr clean up

* chore: clean up

* chore: translations

* chore: pr feedback addressed

* chore updated hook

* chore: Fixed useMapStyle() to incorporate mapServer

* chore: update useMapStyle

* chore: cleaned up hook to be declarative(?)

* chore: clean up

* chore: fixed e2e testing with back button

* chore:clean up of translations

* chore: updated typing

* useMapStyle alternative approach

* chore: updates useMapStyle()

* chore: update hook

* chore: updated modals

* chore: take out superfluos text

* fix: e2e test, newDraft undefined not null

* chore: PR fixes

* chore: merge issues clean up

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

ErikSin committed:
Revert "Map selector (#966)" (#972)

This reverts commit 31f6a008bc856f44c48a495a1f68f39e17462362.

achou11 committed:
chore: set up initial map server integration (#969)

ErikSin committed:
Map selector (#973)

* feat: Added map settings

* chore: Translations

* feat: Created UX Pill Component

* Update en.json

* chore: Added Offline Area list

* chore: translations

* chore:completed BG Maps Screens

* chore: translations

* chore:handle import press

* chore:update Spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* Chore: Casing

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: spelling

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

* chore: PR fixes

* chore: translations

* chore: fixed flicker

* chore: modal components inline

* chore: updated Translations

* chore: hidden under feature flag

* chore: cleaned up feature flag

* chore: updated nav state feature flags

* feat: added experiments hook

* chore: feature flag clean up

* chore: clean up typing

* chore: clean up appContainer

* chore: clean up

* chore: cleaned up linting

* Added Background Map Selector to Map View

* chore: translations

* chore:Pr clean up

* chore: clean up

* chore: translations

* chore: pr feedback addressed

* chore updated hook

* chore: Fixed useMapStyle() to incorporate mapServer

* chore: update useMapStyle

* chore: cleaned up hook to be declarative(?)

* chore: clean up

* chore: fixed e2e testing with back button

* chore:clean up of translations

* chore: updated typing

* useMapStyle alternative approach

* chore: updates useMapStyle()

* chore: update hook

* chore: updated modals

* chore: take out superfluos text

* fix: e2e test, newDraft undefined not null

* chore: PR fixes

* chore: merge issues clean up

* chore: declarative styling

* chore: cleanup

* chore: revert previous behaviour for online style check

* chore: useMapStyle Typings

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>
Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

gmaclennan committed:
chore: simplify nodejs startup & fix bugsnag (#974)

* chore: update nodejs-mobile

* chore: startup node with config, fix bugsnag

* fix patch

* Add missing mock for startWithArgs method

* fix all the errors (oops)

* fix frontend tests

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

gmaclennan committed:
chore: bugsnag error reporting improvements (#975)

- Synchronize release stage between frontend and backend
- Upload react native source maps
- Use bugsnag react error boundary
- Track feature flags with bugsnag

achou11 committed:
Merge branch 'deploy' into develop

sethvincent committed:
add error listener to kappa-osm instance

sethvincent committed:
Merge pull request #983 from digidem/osm-error-listener

add error listener to kappa-osm instance

achou11 committed:
Merge branch 'deploy' into develop

achou11 committed:
Merge branch 'deploy' into develop

Essentially reverts cd0204ca1e8444aae7dda94ec7f2aa73896561e0 since it is
no longer applicable given the refactored Bugsnag setup

gmaclennan committed:
chore: add Andrew to git-secret

achou11 committed:
chore: git ignore random_seed file from git secret (#990)

ErikSin committed:
React nav upgrade (#992)

* refactor: upgrade react nav to v6

* refactor: added useSetHeader() hook

* chore: updated headers

* chore: update custom header

* chore: updated param of useSetHeader to include messageDescriptor

* chore: removed react-nav v3

* chore: update nav state persistance

* chore: update pill for e2e Tests

* fix: e2e tests

* chore: update ICCA navigation

* chore: took out outdated react-native dependencies

* chore: clean up app container

* chore: e2e tests adjustments

* chore: clean up deeplinking

* fix: e2e tests: discard observation popup

* chore: fix background color

* chore: update nav key

* test: update front end test

* convert CategoryChooser to ts

* update package-lock

* fix TS for observation screens

* convert ObservationDetails to TS

also removes the QuestionNavBar component, which was unused

* convert FormattedData to TS

* remove comment in FormattedData

* chore: address PR feedback

* chore: revert back to app stack

* chore: clean up app container and Modals

* chore: optimize re-renders

* chore: change file name in git for CI

* fix: icca variant compiling

* fix: back button spacing

* chore: took out nested navigator

* chore: added typesafety to screen names

* chore: explicit screen elements

* Update src/frontend/hooks/useIsFullyFocused.ts

Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

* fix: typing from "any" to "unknown"

Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

* chore: pr clean up

* fix: back button animation touch point

* chore: optimize header

* chore cleanup

* chore: took out useSetHeader()

* chore: update depdencies

* chore: cleaned up unused dependencies

* chore: icca header

* chore: fix lintinr

Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

* chore: pr clean up

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>
Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

rudokemper committed:
Add Discord link to README.MD (#1008)

ErikSin committed:
App passcode screens (#1000)

* App passcode screens.

Under a dev feature flag

* chore: translations

* chore: clean up unused filed

* cleanup pr

* styling

ErikSin committed:
Kill passcode screens (#1012)

* feat: App passcode screens.

Under a dev feature flag

* chore: Added Kill Passcode Screens

* chore: translations

* took out paper key

* chore: update translations

* chore: clean up

* Update KillPasscode.tsx

achou11 committed:
chore: update navigation call back to map screen (#1014)

achou11 committed:
Update map server integration (#1007)

Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

ErikSin committed:
App passcode state management (#1013)

* feat: App passcode screens.

Under a dev feature flag

* chore: clean up

* feat: App passcode screens.

Under a dev feature flag

* chore: translations

* chore: added security context

* chore: translations

* feat: added confirmation sheet ref

* chore: translations

* chore: focus passcode input

* updated architecture

* change wording

* Update en.json

* translations

* Update index.tsx

* update security context

* input passcode screen

* change file name

* updated security context

* Update ObscurePasscode.tsx

* Update SecurityContext.tsx

* colocate auth state with obscure code

* update passcode to not intersect obscure code

luandro committed:
feat: add colored border to categories in category chooser (#1004)

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

ErikSin committed:
Clean up map manager implementation (#1023)

* feat: In Experiments: Added Offline Map Server

* chore: translations

* chore: disable deletion of default map

* chore: clean up and prep for QA

* chore: translations

* chore: update description

* chore: translations

* chore: update package json after rebase

* chore: cleaned up implementation

* chore: clean up usePersistedState use

* chore: clean up loading state

* chore: take out unnecessary bgmapcard code

* chore: revert offline areas

* chore: translations

* chore: Pr clean up

* chore: translations

* fix: modal not closing after import

* chore: added padding to mapview bottom sheet

* chore: simplyfied logic

* fix: no styleId error

* chore: took out unused param/arg

ErikSin committed:
feat: app authentication and obscure code (#1021)

* chore: took app passcode out of dev experiments

* chore:translations

* chore: linting

* chore: removed setToUnauthenticated

* fix: change auth nav structure

* chores: updated auth flow

* chore: passed removeListener to security Context

* chore: adjust some styling

* fix: authentication state logic

* fix: move auth navigation to AppNavigator

This avoids potential race condition when the navigation ref is not
initialized when the first useEffect runs

* chore: flatten screen groups

* feat: enable auth in ICCA variant

* fix: ui hidden error messages

* fix: get rid of security header

* fix: made entire obscure passcode panel touchable

* chore: update with copy

* chore: translations

Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

ErikSin committed:
Updated copy for map manager (#1032)

ErikSin committed:
Chore: add bottom sheet to confirm passcode (#1030)

* chore: took app passcode out of dev experiments

* chore:translations

* chore: linting

* chore: removed setToUnauthenticated

* fix: change auth nav structure

* chores: updated auth flow

* chore: adjust some styling

* fix: authentication state logic

* fix: move auth navigation to AppNavigator

This avoids potential race condition when the navigation ref is not
initialized when the first useEffect runs

* fix: ui hidden error messages

* fix: made entire obscure passcode panel touchable

* chore: update with copy

* chore: translations

* chore: added bottom sheet to confirm passcode

* chore: added description

* chore: fix rebase

* chore: added line break

* chore: translations

* chore: update padding to remain consistent with design spec

Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

achou11 committed:
fix: fix build issues caused by jcenter shutdown (#1035)

achou11 committed:
chore: fix default background color for navigation screen content (#1037)

gmaclennan committed:
chore: fix export command in CONTRIBUTING.md

ErikSin committed:
chore: updated wcmc copy (#1038)

gmaclennan committed:
fix: Bump react-native version to fix build failures (#1040)

Co-authored-by: Andrew Chou <andrewchou@fastmail.com>

gmaclennan committed:
fix: Only show languages with translated strings (#1042)

gmaclennan committed:
chore: Build-time warning if language name is missing (#1044)

* fix: Warn if language name is missing

* Write error from build translations script...

not polyfill build script

gmaclennan committed:
chore: convert IntlContext to Typescript (#1045)

ErikSin committed:
feat: deletion of offline map (#1029)

* feat: added page to start bg map delete

* chore: added Map

* feat: added bottom sheet modal for deletion

* chore: pr review clean up

* chore: updated zoom level search

* chore: convert promise all to all settled

* fix: all settled

* chore:clean up tilesetIds Array

* chore: update bottom sheet to bottom sheet modal

* chore: fix rebase navigation changes

* chore:styling of delete button

* chore: updated zoom level to zoom range

* chore: translations

* chore: removed zoom

ErikSin committed:
chore: remove background maps (#1051)

* chore: remove background map

* chore: added loading check

* chore: update merge state

digidem-bot committed:
New Crowdin updates (#991)

* New translations en.json (French)

* New translations en.json (Thai)

* New translations en.json (Spanish)

* New translations en.json (Indonesian)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (French)

* New translations en.json (French)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Vietnamese)

* New translations en.json (Vietnamese)

* New translations en.json (Thai)

* New translations en.json (Vietnamese)

* New translations en.json (Vietnamese)

* New translations en.json (Tamil)

* New translations en.json (Tamil)

* New translations en.json (Tamil)

* New translations en.json (Tamil)

* New translations en.json (Sinhala)

* New translations en.json (Sinhala)

* New translations en.json (Sinhala)

* New translations en.json (Tamil)

* New translations en.json (Sinhala)

* New translations en.json (Tamil)

* New translations en.json (Sinhala)

* New translations en.json (Tamil)

* New translations en.json (Sinhala)

* New translations en.json (Tamil)

* New translations en.json (Sinhala)

* New translations en.json (Sinhala)

* New translations en.json (Sinhala)

* New translations en.json (Sinhala)

* New translations en.json (Sinhala)

* New translations en.json (Tamil)

* New translations en.json (Sinhala)

* New translations en.json (Tamil)

* New translations en.json (Sinhala)

* New translations en.json (Tamil)

* New translations en.json (Tamil)

* New translations en.json (Sinhala)

* New translations en.json (Tamil)

* New translations en.json (Tamil)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Santali)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Thai)

* New translations en.json (Sinhala)

* New translations en.json (Sinhala)

* New translations en.json (Sinhala)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Vietnamese)

* New translations en.json (Vietnamese)

* New translations en.json (Thai)

* New translations en.json (French)

* New translations en.json (Sinhala)

* New translations en.json (Thai)

* New translations en.json (Spanish)

* New translations en.json (Indonesian)

* New translations en.json (Ogiek)

* New translations en.json (Swahili, Kenya)

* New translations en.json (Ogiek)

* New translations en.json (Swahili)

* New translations en.json (Swahili)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Hausa)

* New translations en.json (Fula)

* New translations en.json (French)

* New translations en.json (Burmese)

* New translations en.json (Sranan Tongo)

* New translations en.json (Nepali)

* New translations en.json (Sinhala)

* New translations en.json (Spanish)

* New translations en.json (Thai)

* New translations en.json (Tamil)

* New translations en.json (Khmer)

* New translations en.json (Indonesian)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Vietnamese)

* New translations en.json (Dutch)

* New translations en.json (German)

* New translations en.json (French)

* New translations en.json (Sinhala)

* New translations en.json (Spanish)

* New translations en.json (Thai)

* New translations en.json (Tamil)

* New translations en.json (Vietnamese)

* New translations en.json (French)

* New translations en.json (Sinhala)

* New translations en.json (Spanish)

* New translations en.json (Thai)

* New translations en.json (Tamil)

* New translations en.json (Vietnamese)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

* New translations en.json (Sinhala)

* New translations en.json (Thai)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Vietnamese)

* New translations en.json (Vietnamese)

* New translations en.json (Nepali)

* New translations en.json (Sinhala)

* New translations en.json (Spanish)

* New translations en.json (Thai)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Vietnamese)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Spanish)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Spanish)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Vietnamese)

* New translations en.json (Vietnamese)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Spanish)

* New translations en.json (Portuguese, Brazilian)

* New translations en.json (Vietnamese)

* New translations en.json (Nepali)

* New translations en.json (Nepali)

Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>
Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

ErikSin committed:
chore(release): Prepare release 5.5.0

ErikSin committed:
chore(release): update CHANGELOG.md

ErikSin committed:
fix: app passcode stack available in ICCA

ErikSin committed:
fix: photos modal opens photo in observation edit

ErikSin committed:
fix: zoom on non-offline maps

ErikSin committed:
fix: clicking back observation details works as intended

ErikSin committed:
fix: after details, navigates to edit, not new observations

ErikSin commented:
@reviewer: [c37adb4](https://github.com/digidem/mapeo-mobile/pull/1053/commits/c37adb458bc6751d7fdb017f511ae5d05206c8db) fixes passcode stack not being available in ICCA variant

ErikSin commented:
@Reviewer [c9ca0c1](https://github.com/digidem/mapeo-mobile/pull/1053/commits/c9ca0c1ca248fa41f1f9f25fc5676132630d4a67) fixes photo modal not correctly opening photo.

See [notion issue](https://www.notion.so/digidem/When-editing-saved-observations-Photo-view-screen-appears-all-black-and-includes-Delete-Image-butto-f655fe813d0e4e62a282dd37bf594a22) for reference

ErikSin commented:
@Reviewer [88ee1ee](https://github.com/digidem/mapeo-mobile/pull/1053/commits/88ee1eea5cdf497136132c5fa8d953561b441b1e) fixes auto zoom not working as intended

See [notion issue ](https://www.notion.so/digidem/Default-zoom-level-on-home-Map-screen-far-lower-than-standard-Mapeo-build-Intentional-31a84df6a17a4460b1215201d4cb7150)for reference

achou11 commented on src/frontend/screens/AppPasscode/EnterPassToTurnOff.tsx:
```diff
@@ -1,16 +1,16 @@
 import * as React from "react";
-import { defineMessage } from "react-intl";
+import { defineMessages } from "react-intl";
```
need to do the same thing for `LeaveProjectInitial.tsx`

ErikSin replied:
OH, yes, its on my local device, I thought i pushed it with this

ErikSin replied:
https://github.com/digidem/mapeo-mobile/pull/1053/commits/c8f9fc44cdfbf9ad9b949126a06a11cac8b5c8cf

ErikSin commented:
@Reviewer [bf1cf65](https://github.com/digidem/mapeo-mobile/pull/1053/commits/bf1cf65d9013c8afbd83afb12b29b4105b9da956) fixes observation detail going to the previous question when pressing back, instead of closing observation detail.

See [Notion issue](https://www.notion.so/digidem/button-in-Details-screens-returns-to-New-observation-or-Edit-observation-screen-rather-than-previou-0bc3421d2abb4430a386234d5895f1ca) for reference

ErikSin commented:
@Reviewer [9b0860b](https://github.com/digidem/mapeo-mobile/pull/1053/commits/9b0860bb33de58788a9d454d2a92b216a63367e8) fixes when completing the observation detail questions, while editing an observation, the edit view title appears as `edit observation` NOT `new observation`.

See [Notion issue](https://www.notion.so/digidem/When-editing-saved-observations-clicking-DONE-after-Details-screens-returns-to-New-observation-scre-c50a0dfc30ad449d9b5219b221249238) for reference.

ErikSin commented on src/frontend/screens/ObservationDetails/index.tsx:
```diff
@@ -35,10 +37,24 @@ const ObservationDetails = ({
   route,
 }: NativeRootNavigationProps<"ObservationDetails">) => {
   const [{ preset }] = useDraftObservation();
-  const current: number = +route.params.question;
```
I believe this was a typo that made it into the code. Adding the current question number to itself does not make sense. the user is navigated to a NEW instance of this screen every time a new question appears, with the question param being updated.

achou11 replied:
i don't think this is a typo actually. it's a (unintentionally?) clever way of converting a potential string into a number 😅 e.g. `const foo = +"1"` results in `foo` being equal to `1`. probably not necessary given how we have react native navigation set up though

> Adding the current question number to itself does not make sense

That's not what this is doing. it would be written as `const current: number += route.params.question` if that were the case, which is invalid

ErikSin replied:
ahh yes very clever. but yes, you are correct. With the new navigation we can pass all primitive types, and they will be extracted as the correct type

ErikSin committed:
chore: localized messages not being extracted (#1065)

gmaclennan committed:
chore: don't run install scripts on backend build (#970)

This is just a DX issue to save time and unnecessary downloads during the
backend build. The install / postinstall scripts all seem to be for
downloading or compiling native builds, but done during install this will be
the native builds for the dev machine arch. We will be rebuilding these anyway
for mobile, so no need to run these scripts. Ensure we start in the right place

ErikSin committed:
chore: zoom on fallback map

achou11 committed:
fix: upgrade android target sdk to fulfill google play store requirements (#1059)

* change target sdk to 31

* required code changes in order to target 31

* mapbox madness

* add comment about why buildToolsVersion has to be 30

* update contributing docs

* update app/build.gradle to use java 11

* remove compileOptions in app/build.gradle

caused detox build command to fail. runs fine without it now

* fix: use fixed docker container

Co-authored-by: Gregor MacLennan <gmaclennan@digital-democracy.org>

achou11 committed:
chore: use v2 of docker android for e2e test CI

ErikSin committed:
chore: update location button props
